### PR TITLE
[WIP] Feature/skillnode deselection cascades

### PR DESCRIPTION
**Pull request by juanmanual**
Implements the cascading deselection discussed in #16

Todo: Implement stubbed test behavior

### CODE REVIEW

**Comment by juanmanual** [comment]
I mean to open this as a draft. Anyways, there is currently a visual bug with this implementation where the skill nodes that are made locked due to cascade appear selectable until you click on them, and then their children will have this effect.

File: src/components/SkillNode.tsx [hunk 1]
```diff
@@ -119,7 +121,8 @@ class SkillNode extends React.Component<Props, State> {
     const parentNodeIsSelected =
       !parentNodeId || parentState === SELECTED_STATE;
 
-    if (nodeState === UNLOCKED_STATE && !parentNodeIsSelected) {
+    if (nodeState === SELECTED_STATE && !parentNodeIsSelected) {
+      this.context.decrementSelectedSkillCount();
```
Comment (andrico1234) [inline src/components/SkillNode.tsx]: I've only tested quickly, but you still need to check whether the state is unlocked when it updated. If you replace the code with this, add some tests then it should be fine!

```
    if (nodeState === SELECTED_STATE && !parentNodeIsSelected) {
      this.context.decrementSelectedSkillCount();
      return this.updateState(LOCKED_STATE);
    }

    if (nodeState === UNLOCKED_STATE && !parentNodeIsSelected) {
      return this.updateState(LOCKED_STATE);
    }

```

**Comment by andrico1234** [comment]
So I've worked on a feature and I'm going to implement this along with it! Do you don't need to do any more work in this PR!

I'll credit you as a contributor in the Readme! 😊

**Comment by juanmanual** [comment]
Okay sweet

**Pull request closed without merging**